### PR TITLE
Bug fix & evol (v0.0.5)

### PR DESCRIPTION
**Pull request by ThibautRappet**
bug fix & new features 

mainly : 

- post message with delivery delay support
- force retry strategy support 

some points need MerQure & MerQure.Tools update

### CODE REVIEW

**Comment by ThibautRappet** [comment]
@BriceFrancois un avis avant que je merge ?

**Comment by BriceFrancois** [comment]
@ThibautRappet je regarde dans la journée

File: src/MerQure.Tools/Buses/Bus.cs [hunk 1]
```diff
@@ -18,39 +18,44 @@ public Bus(Publisher<T> producer, RetryConsumer<T> consumer)
...
         }
 
-        public void Consume(Channel channel, EventHandler<T> callback)
+        public void OnMessageReceived(Channel channel, EventHandler<T> callback)
         {
             Consumer.Consume(channel, callback);
         }
 
-        public void Publish(Channel channel, T message)
+        public void Publish(Channel channel, T message, bool applyDeliveryDeplay = false)
```
Comment (BriceFrancois) [inline src/MerQure.Tools/Buses/Bus.cs]: applyDeliveryDeplay ?

File: src/MerQure.Tools/Buses/Bus.cs [hunk 2]
```diff
@@ -18,39 +18,44 @@ public Bus(Publisher<T> producer, RetryConsumer<T> consumer)
...
 
-        public void RejectDeliveredMessage(Channel channel, T delivredMessage)
+        public void RejectDeliveredMessage(Channel channel, T deliveredMessage)
         {
-            Consumer.RejectDeliveredMessage(channel, delivredMessage);
+            Consumer.RejectDeliveredMessage(channel, deliveredMessage);
         }
 
-        public void ApplyRetryStrategy(Channel channel, T delivredMessage)
+        public MessageInformations ApplyRetryStrategy(Channel channel, T deliveredMessage)
```
Comment (BriceFrancois) [inline src/MerQure.Tools/Buses/Bus.cs]: What will be done ? 
Why not just naming it ```Retry()``` (which strategy ?)

File: src/MerQure.Tools/Buses/Bus.cs [hunk 3]
```diff
@@ -18,39 +18,44 @@ public Bus(Publisher<T> producer, RetryConsumer<T> consumer)
             Consumer = consumer;
         }
 
-        public void AcknowlegdeDelivredMessage(Channel channel, T delivredMessage)
+        public void AcknowlegdeDeliveredMessage(Channel channel, T deliveredMessage)
```
Comment (BriceFrancois) [inline src/MerQure.Tools/Buses/Bus.cs]: Channel should on Bus, no ?

File: src/MerQure.Tools/Buses/Bus.cs [hunk 4]
```diff
@@ -18,39 +18,44 @@ public Bus(Publisher<T> producer, RetryConsumer<T> consumer)
...
         }
 
-        public void SendDelivredMessageToErrorBus(Channel channel, T delivredMessage)
+        public void SendDeliveredMessageToErrorBus(Channel channel, T deliveredMessage)
         {
-            Consumer.SendToErrorExchange(channel, delivredMessage);
+            Consumer.SendToErrorExchange(channel, deliveredMessage);
+        }
+
+        public MessageInformations PublishForceRetryAttemptNumber(Channel channel, T message, int attemptNumber)
```
Comment (BriceFrancois) [inline src/MerQure.Tools/Buses/Bus.cs]: ```ForceRetry(T message, int attemptsMaxNumber)``` ?

**Pull request merged**